src/enumo/workload.rs

```diff
@@ -157,7 +157,14 @@ impl Workload {
     }
 
     pub fn append(self, workload: impl Into<Workload>) -> Self {
-        Workload::Append(vec![self, workload.into()])
+        match self {
```

Comment:
What if workload is an append? Wouldn't you want to consolidate with that one as well?